### DISCUSSION TITLE
Bulk transfer polling - question

### DISCUSSION THREAD

**Discussion by szoshi** [pyusb/pyusb#512] [Q&A]
I am trying to maximize the throughput of a device doing high speed bulk transfers. I notice using a protocol analyzer that the host checks for data just three or four times per microframe. I can confirm that there is no other traffic on the USB bus besides that which goes to my test device. I have read that theoretically, there can be up to 13 transfers per microframe. In practice, this number should be closer to 10. My host seems to check just 4 times per microframe. Is there a way in the underlying libusb to configue this? Can this be influenced from pyusb? I am on a windows machine. Does someone have a different result from a Linux machine?

**Comment by mcuee** [comment]
The firmware plays a part as well.

You can also refer to this discussion and use a large transfer size to see if that helps.
https://github.com/pyusb/pyusb/discussions/507
